sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -1019,104 +1006,108 @@ public void startCopyFilePermissionFormat(FilePermissionFormat filePermissionFor
             .expectComplete()
             .verify(Duration.ofMinutes(1));
 
-        FileSmbProperties properties
-            = Objects.requireNonNull(primaryFileAsyncClient.getProperties().block()).getSmbProperties();
-
-        FileShareTestHelper.compareDatesWithPrecision(properties.getFileCreationTime(),
-            smbProperties.getFileCreationTime());
-        FileShareTestHelper.compareDatesWithPrecision(properties.getFileLastWriteTime(),
-            smbProperties.getFileLastWriteTime());
-        assertEquals(properties.getNtfsFileAttributes(), smbProperties.getNtfsFileAttributes());
+        StepVerifier.create(primaryFileAsyncClient.getProperties()).assertNext(properties -> {
+            FileSmbProperties smbProps = properties.getSmbProperties();
+            FileShareTestHelper.compareDatesWithPrecision(smbProps.getFileCreationTime(),
+                smbProperties.getFileCreationTime());
+            FileShareTestHelper.compareDatesWithPrecision(smbProps.getFileLastWriteTime(),
+                smbProperties.getFileLastWriteTime());
+            assertEquals(smbProps.getNtfsFileAttributes(), smbProperties.getNtfsFileAttributes());
+        }).verifyComplete();
     }
 
     @Test
     public void startCopyWithOptionsLease() {
-        primaryFileAsyncClient.create(1024).block();
-        String sourceURL = primaryFileAsyncClient.getFileUrl();
-        String leaseId = createLeaseClient(primaryFileAsyncClient).acquireLease().block();
-        ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
+        Mono<ShareFileInfo> createFileMono = primaryFileAsyncClient.create(1024);
+        Mono<String> leaseIdMono = createFileMono.then(createLeaseClient(primaryFileAsyncClient).acquireLease());
 
-        ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
+        String sourceURL = primaryFileAsyncClient.getFileUrl();
 
-        PollerFlux<ShareFileCopyInfo, Void> poller
-            = setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
-        StepVerifier.create(poller)
-            .assertNext(it -> assertNotNull(it.getValue().getCopyId()))
-            .expectComplete()
-            .verify(Duration.ofMinutes(1));
+        StepVerifier.create(leaseIdMono.flatMapMany(leaseId -> {
+            ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
+            ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
+            PollerFlux<ShareFileCopyInfo, Void> poller
+                = setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
+            return poller.take(1).flatMap(response -> {
+                assertNotNull(response.getValue().getCopyId());
+                return Mono.just(response);
+            });
+        })).expectNextCount(1).verifyComplete();
```

Comment:
We can also make use of returning just the PollerFlux here instead of doing anything extra to it:

```
@Test
    public void startCopyWithOptionsLease() {
        String sourceURL = primaryFileAsyncClient.getFileUrl();
        Mono<String> leaseIdMono
            = primaryFileAsyncClient.create(1024).then(createLeaseClient(primaryFileAsyncClient).acquireLease());

        StepVerifier.create(leaseIdMono.flatMapMany(leaseId -> {
            ShareRequestConditions conditions = new ShareRequestConditions().setLeaseId(leaseId);
            ShareFileCopyOptions options = new ShareFileCopyOptions().setDestinationRequestConditions(conditions);
            return setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
        })).assertNext(it -> assertNotNull(it.getValue().getCopyId())).expectComplete().verify(Duration.ofMinutes(1));
    }
```